graphql/resolvers.go

```diff
@@ -1417,6 +1417,30 @@ func (r *mutationResolver) RemovePublicKey(ctx context.Context, keyName string)
 	return myPublicKeys, nil
 }
 
+func (r *mutationResolver) UpdatePublicKey(ctx context.Context, targetKeyName string, updateInfo PublicKeyInput) ([]*restModel.APIPubKey, error) {
+	if !doesPublicKeyNameAlreadyExist(ctx, targetKeyName) {
+		return nil, InputValidationError.Send(ctx, fmt.Sprintf("Error updating public key. The target key name, %s, does not exist.", targetKeyName))
+	}
+	if updateInfo.Name != targetKeyName && doesPublicKeyNameAlreadyExist(ctx, updateInfo.Name) {
+		return nil, InputValidationError.Send(ctx, fmt.Sprintf("Error updating public key. The updated key name, %s, already exists.", targetKeyName))
+	}
+	err := verifyPublicKey(ctx, updateInfo)
+	if err != nil {
+		return nil, err
+	}
+	usr := route.MustHaveUser(ctx)
+	err = usr.DeletePublicKey(targetKeyName)
```

Comment:
Instead of doing this in 2 queries, can you create 1 to update the key so that it's atomic? Alternatively I think it's totally fine to just support delete/add a key since needing to update an existing key should be fairly rare

Reply:
sure thing 👍